hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -899,11 +900,26 @@ public String getMultipartKeyFSO(String volume, String bucket, String key, Strin
     final long volumeId = getVolumeId(volume);
     final long bucketId = getBucketId(volume,
             bucket);
-    long parentId =
-            OMFileRequest.getParentID(volumeId, bucketId, key, this);
-
-    String fileName = OzoneFSUtils.getFileName(key);
+    long parentId;
+    try {
+      parentId = OMFileRequest.getParentID(volumeId, bucketId, key, this);
```

Comment:
I am going over the code in a bit more detail but why not use the exception handling way to calculate the `parentId` as the default way?

Reply:
This was my suggestion since there might be some codes that depends on the fact `getParentID` implementation (i.e. parent directory exists). 

I'm fine to use the updated way as long as there are no regressions found.

Reply:
Yeah. I just follow @ivandika3 suggestion, which is safer. And I tested if I just use the exception handling way to do it, then `OMKeyCreateRequestWithFSO#getDBMultipartOpenKey` method will throw exception when the MPU is aborted. And this method is used in multiple places. Of course, we can update all these places. I am just not sure if it is safe. Just try to limit the scope of this PR. Pls let me know your toughts